src/session.ts

```diff
@@ -60,13 +60,16 @@ export class DebugSession implements IDebugger.ISession {
       this._client.iopubMessage.disconnect(this._handleEvent, this);
     }
 
-    this._client = client;
-
     if (client) {
+      this._client = client;
       this._client.iopubMessage.connect(this._handleEvent, this);
     }
   }
 
+  get debuggedClients(): Set<string> {
```

Comment:
Does this mean that a debug session should keep track of multiple clients?

Reply:
this was only my suggestion. For now, we still tracking all clients which are have supported kernel.

Reply:
Not sure yet the `session` is the preferred place to store such information.

But it would be fine to refactor that in a follow-up PR.